community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/planner/logical/plans/LogicalPlan.scala

```diff
@@ -99,6 +99,22 @@ abstract class LogicalPlan
   def mapExpressions(f: (Set[IdName], Expression) => Expression): LogicalPlan
 
   def isLeaf: Boolean = lhs.isEmpty && rhs.isEmpty
+
+  override def toString = {
+    val leafs = lhs.toSeq ++ rhs.toSeq
```

Comment:
I don't get it these are not necessarily leafs (nor leaves :) )? Do you mean `children`?